exchanges/request/request.go

```diff
@@ -247,36 +248,50 @@ func (r *Requester) DoRequest(req *http.Request, method, path string, headers ma
 		log.Printf("%s exchange request path: %s requires rate limiter: %v", r.Name, path, r.RequiresRateLimiter())
 	}
 
-	resp, err := r.HTTPClient.Do(req)
+	var timeoutError error
+	for i := 0; i < 4; i++ { // 3 attempts for extra requests due to client timeout
```

Comment:
Create a request var called DefaultTimeoutRetryAttempts and set that to 3, in the for loop change it to i < DefaultTimeoutRetryAttempts +1

Engine refactor will include code to make this overridable and optional based on user preferences.

Reply:
Okay will do

Reply:
Could we go with `i <= X` rather than `i < (X +1)` ?

Reply:
Destinction of retry attempts val and primary request is better IMO.

Reply:
Yeah if we pass in 0 then it won't do a request.